src/daft-io/src/s3_like.rs

```diff
@@ -798,6 +798,30 @@ impl ObjectSource for S3LikeSource {
             io_stats,
         )
         .await
+        .map(
+            // Filter the result stream for any size-0 File entries that end with "/"
+            // These are usually used to demarcate "empty folders", since S3 is not really a filesystem
+            // However they can lead to unexpected globbing behavior since most users do not expect them to exist
+            |result_stream| {
+                result_stream
+                    .filter_map(|fm| async move {
```

Comment:
I cannot for the life of me get `.filter(...)` to work! 😭 

`.filter` receives `fm` as a reference, which seems to cause a whole bunch of lifetime issues

Reply:
Moved logic to shared `fn glob` utility, inside of the `stream!` so we don't have to use the `StreamExt::filter` function anymore